### PR TITLE
bump to go 1.23

### PR DESCRIPTION
**Pull request by techknowlogick**

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by yp05327** [comment]
devcontainer should also be updated.

It seems 1.23 has been supported but I got image not found error.
https://github.com/devcontainers/images/pull/1160
image

**Comment by techknowlogick** [comment]
@yp05327 it takes time for the image to build and push.

File: Makefile [hunk 1]
```diff
@@ -23,12 +23,12 @@ SHASUM ?= shasum -a 256
...
 COMMA := ,
 
-XGO_VERSION := go-1.22.x
+XGO_VERSION := go-1.23.x
 
 AIR_PACKAGE ?= github.com/air-verse/air@v1
 EDITORCONFIG_CHECKER_PACKAGE ?= github.com/editorconfig-checker/editorconfig-checker/cmd/editorconfig-checker@2.7.0
 GOFUMPT_PACKAGE ?= mvdan.cc/gofumpt@v0.6.0
-GOLANGCI_LINT_PACKAGE ?= github.com/golangci/golangci-lint/cmd/golangci-lint@v1.59.0
+GOLANGCI_LINT_PACKAGE ?= github.com/golangci/golangci-lint/cmd/golangci-lint@v1.60.1
```
Comment (silverwind) [inline Makefile]:
```suggestion
GOLANGCI_LINT_PACKAGE ?= github.com/golangci/golangci-lint/cmd/golangci-lint@v1.60.3
```

File: Makefile [hunk 2]
```diff
@@ -23,12 +23,12 @@ SHASUM ?= shasum -a 256
 HAS_GO := $(shell hash $(GO) > /dev/null 2>&1 && echo yes)
 COMMA := ,
 
-XGO_VERSION := go-1.22.x
+XGO_VERSION := go-1.23.x
 
 AIR_PACKAGE ?= github.com/air-verse/air@v1
 EDITORCONFIG_CHECKER_PACKAGE ?= github.com/editorconfig-checker/editorconfig-checker/cmd/editorconfig-checker@2.7.0
 GOFUMPT_PACKAGE ?= mvdan.cc/gofumpt@v0.6.0
```
Comment (silverwind) [inline Makefile]:
```suggestion
GOFUMPT_PACKAGE ?= mvdan.cc/gofumpt@v0.7.0
```

**Comment by silverwind** [comment]
More lint fixed needed, otherwise lgtm.

**Review by a1012112796 — APPROVED** [review]

**Pull request merged**